### PR TITLE
Fix/cppcheck 4

### PR DESCRIPTION
**Pull request by plbossart**
As usual cppcheck reports mostly false positives, except that this one it found what looks like a real bug....

Cleanup and make false positives go away to that we can keep looking for *REAL* problems.

### CODE REVIEW

**Review by RanderWang — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

File: sound/soc/sof/intel/hda-ipc.c [hunk 1]
```diff
@@ -160,11 +161,10 @@ irqreturn_t hda_dsp_ipc4_irq_thread(int irq, void *context)
 			spin_unlock_irq(&sdev->ipc_lock);
 		} else {
 			/* Notification received */
-			struct sof_ipc4_msg data = {{ 0 }};
 
 			data.primary = primary;
 			data.extension = extension;
-			sdev->ipc->msg.rx_data = &data;
+			sdev->ipc->msg.rx_data = &null_data;
```
Comment (bardliao) [inline sound/soc/sof/intel/hda-ipc.c]: @plbossart This is not a null data. We set data.primary and extension above. And it will lead to build error

Comment (ujfalusi) [inline sound/soc/sof/intel/hda-ipc.c]: a bit silly that cppcheck got this one wrong...

**Review by ujfalusi — APPROVED** [review]

**Review by bardliao — CHANGES_REQUESTED** [review]
Found build error. See comment

File: sound/soc/sof/intel/hda-ipc.c (hunk 1 above)
Comment (ujfalusi) [reply sound/soc/sof/intel/hda-ipc.c]: for the record, I'm using `cppcheck` locally with these parameters and it is not complaining:
`--platform=unix64 --force --max-configs=1024 --inconclusive --enable=all --suppress=variableScope --suppress=shiftTooManyBitsSigned --suppress=arithOperationsOnVoidPointer --suppress=bitwiseOnBoolean`
Should I extend the parameters (without options cppcheck is all green for hda-ipc.c)?

Comment (ujfalusi) [reply sound/soc/sof/intel/hda-ipc.c]: and this is valid for the previous change as well at line 79

**Review by ujfalusi — CHANGES_REQUESTED** [review]

File: sound/soc/sof/intel/cnl.c [hunk 2]
```diff
@@ -72,12 +73,10 @@ irqreturn_t cnl_ipc4_irq_thread(int irq, void *context)
 			spin_unlock_irq(&sdev->ipc_lock);
 		} else {
 			/* Notification received */
-			struct sof_ipc4_msg data = {{ 0 }};
+			null_data.primary = primary;
+			null_data.extension = extension;
```
Comment (ujfalusi) [inline sound/soc/sof/intel/cnl.c]: If we are renaming it, can we name it something like `rx_data` or `notification`? `null_data` does not make much sense here.

File: sound/soc/sof/intel/hda-ipc.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/intel/hda-ipc.c]: I also use the same parameters @ujfalusi 

````
cppcheck --platform=unix64 --force --max-configs=1024 --inconclusive --enable=all --suppress=variableScope --suppress=shiftTooManyBitsSigned --suppress=arithOperationsOnVoidPointer --suppress=bitwiseOnBoolean sound/soc/sof/
````

cppcheck --version
Cppcheck 2.3

File: sound/soc/sof/intel/cnl.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/sof/intel/cnl.c]: I renamed it notification_data.

File: sound/soc/sof/intel/hda-ipc.c (hunk 1 above)
Comment (ujfalusi) [reply sound/soc/sof/intel/hda-ipc.c]: @plbossart, interesting
```
cppcheck --version
Cppcheck 2.7.4
```
on my side.

**Comment by plbossart** [comment]
Surprising fail on https://sof-ci.01.org/linuxpr/PR3597/build21/devicetest/?model=WHL_UPEXT_HDA_ZEPHYR&testcase=check-pause-resume-capture-10
I seem to have zero luck with UpExtreme boards....

**Comment by plbossart** [comment]
SOFCI TEST

**Review by bardliao — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Review by ujfalusi — APPROVED** [review]

**Pull request merged**